pkg/frontend/querymiddleware/codec_test.go

```diff
@@ -346,17 +346,83 @@ func TestMetricsQuery_MinMaxTime_TransformConsistency(t *testing.T) {
 			initialMetricsQuery: instantRequest,
 			updatedStartTime:    &updatedEndTime,
 			updatedEndTime:      &updatedEndTime,
-			updatedQuery:        "",
 
 			expectedUpdatedMinT: updatedEndTime.UnixMilli(),
 			expectedUpdatedMaxT: updatedEndTime.UnixMilli(),
 			expectedErr:         nil,
 		},
+	} {
+		t.Run(testCase.name, func(t *testing.T) {
+			start := testCase.initialMetricsQuery.GetStart()
+			end := testCase.initialMetricsQuery.GetEnd()
+
+			// apply WithStartEnd
+			newStart := start
+			if testCase.updatedStartTime != nil {
+				newStart = testCase.updatedStartTime.UnixMilli()
+			}
+			newEnd := end
+			if testCase.updatedEndTime != nil {
+				newEnd = testCase.updatedEndTime.UnixMilli()
+			}
```

Comment:
This can now be simplified further: `updatedStartTime` and `updatedEndTime` are now always non-`nil`, so we can drop the conditions here (and make them non-pointer values as well)